src/ecs/resources/input/input_handler.rs

```diff
@@ -36,8 +36,8 @@ pub struct InputHandler {
     pressed_mouse_buttons: SmallVec<[MouseButton; 16]>,
     down_mouse_buttons: SmallVec<[MouseButton; 16]>,
     released_mouse_buttons: SmallVec<[MouseButton; 16]>,
-    mouse_position: Option<(i32, i32)>,
-    previous_mouse_position: Option<(i32, i32)>,
+    mouse_position: Option<(f64, f64)>,
```

Comment:
I definitely vote for a floating point of some format, it is what I used in my engine.

Interestingly I did not know abut mice doing subprecision, apparently my mouse does this, I just noticed it.  ^.^